client/home/lib/welcome/welcomemodal.coffee

```diff
@@ -31,32 +31,33 @@ module.exports = class WelcomeModal extends kd.ModalView
   showCongratulationModal: ->
 
     { appStorageController } = kd.singletons
-
     appStorage = appStorageController.storage "WelcomeSteps-#{globals.currentGroup.slug}"
-    appStorage.fetchValue 'OnboardingSuccessModalShown', (result) ->
```

Comment:
@gokmen not sure but when I debug I had some experience
https://github.com/koding/koding/blob/master/client/app/lib/appstorage.coffee#L72
In fetchValue function, It is called again and again before it returns the callback,
also had same experience with setValue when I pass callback.